src/View/mainpage/DrawROIWindow/Drawing.py

```diff
@@ -277,9 +371,19 @@ def draw_contour_preview(self, list_of_points):
         of points ordered to form a polygon.
         """
         qpoint_list = []
-        for point in list_of_points:
-            qpoint = QtCore.QPoint(point[0], point[1])
-            qpoint_list.append(qpoint)
+        if self.different_sizes:
+            for point in list_of_points:
+                x_arr, y_arr = inv_linear_transform(
+                    point[0], point[1], self.rows, self.cols)
+                for x in x_arr:
+                    for y in y_arr:
+                        qpoint = QtCore.QPoint(x, y)
+                        qpoint_list.append(qpoint)
+        else:
+            for point in list_of_points:
+                qpoint = QtCore.QPoint(point[0], point[1])
+                qpoint_list.append(qpoint)
+
```

Comment:
Hi Linh, its a good thought but not what I intended, inv_linear_transform's purpose is to turn a 1*1 pixel into an n*n pixel to interact with the QImage, so while its possible, I don't think it will change the complexity as i need all n*n pixels generated.